src/OmniSharp.Roslyn.CSharp/Workers/Diagnostics/DiagnosticAnalyzerExtensions.cs

```diff
@@ -0,0 +1,180 @@
+using System.Collections;
+using System.Collections.Immutable;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Options;
+
+namespace OmniSharp.Roslyn.CSharp.Workers.Diagnostics
+{
+    public static class DiagnosticAnalyzerExtensions
+    {
+        private static MethodInfo TryGetMappedOptionsMethod { get; }
+
+        static DiagnosticAnalyzerExtensions()
+        {
+            TryGetMappedOptionsMethod = Assembly.Load(new AssemblyName("Microsoft.CodeAnalysis.Features"))
```

Comment:
is the reflection here necessary? 
if yes, then for consistency it would make sense to use https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Shared/Utilities/ReflectionExtensions.cs

Reply:
I can work on adding an accessor to the ExternalAccess layer to remove this reflection down the road.